src/ks_json.c

```diff
@@ -451,6 +451,64 @@ KS_DECLARE(ks_json_t *) ks_json_enum_next(ks_json_t *item)
 	return item->next;
 }
 
+KS_DECLARE(ks_bool_t) ks_json_replace_item(ks_json_t *parent, ks_json_t **item, ks_json_t *replacement)
+{
+	if ((parent == NULL) || (replacement == NULL) || (item == NULL) || (*item == NULL)) {
+		return KS_FALSE;
+	}
+
+	if (replacement == *item) {
+		*item = NULL;
+
+		return KS_TRUE;
+	}
+
+	replacement->next = (*item)->next;
+	replacement->prev = (*item)->prev;
+
+	if ((*item)->string) {
+		if (replacement->string) {
+			free(replacement->string);
+		}
+
+		replacement->string = strdup((*item)->string);
```

Comment:
Can we avoid using `strdup` as `item` is a walking dead? Can we just steal it's string pointer and null it in the item?